RemoteInput/Plugin/ControlCenter.cxx

```diff
@@ -1089,6 +1125,206 @@ void ControlCenter::process_reflect_array_with_size(jarray array, void* &argumen
+					//fprintf(stdout, "write long[%zu]\n", i);
+					EIOS_Write(response, result[i]);
+				}
+				reflector->getEnv()->ReleaseLongArrayElements(static_cast<jlongArray>(array), result, JNI_ABORT);
+				//reflector->getEnv()->GetLongArrayRegion(static_cast<jlongArray>(array), index, length, static_cast<jlong*>(response));
+			}
+			break;
+
+			case ReflectionArrayType::FLOAT:
+			{
+				jfloat* result = reflector->getEnv()->GetFloatArrayElements(static_cast<jfloatArray>(array), nullptr);
+				for (std::size_t i = 0; i < length; ++i)
+				{
+					//fprintf(stderr, "write float\n");
+					EIOS_Write(response, ReflectionBinaryType::FLOAT);
+					//fprintf(stdout, "write float[%zu]\n", i);
+					EIOS_Write(response, result[i]);
+				}
+				reflector->getEnv()->ReleaseFloatArrayElements(static_cast<jfloatArray>(array), result, JNI_ABORT);
+				//reflector->getEnv()->GetFloatArrayRegion(static_cast<jfloatArray>(array), index, length, static_cast<jfloat*>(response));
+			}
+			break;
+
+			case ReflectionArrayType::DOUBLE:
+			{
+				jdouble* result = reflector->getEnv()->GetDoubleArrayElements(static_cast<jdoubleArray>(array), nullptr);
+				for (std::size_t i = 0; i < length; ++i)
+				{
+					//fprintf(stderr, "write double\n");
+					EIOS_Write(response, ReflectionBinaryType::DOUBLE);
+					//fprintf(stdout, "write double[%zu]\n", i);
+					EIOS_Write(response, result[i]);
+				}
+				reflector->getEnv()->ReleaseDoubleArrayElements(static_cast<jdoubleArray>(array), result, JNI_ABORT);
+				//reflector->getEnv()->GetDoubleArrayRegion(static_cast<jdoubleArray>(array), index, length, static_cast<jdouble*>(response));
+			}
+			break;
+
+			case ReflectionArrayType::STRING:
+			{
+				for (std::size_t i = 0; i < length; ++i)
+				{
+					auto element = reflector->getEnv()->GetObjectArrayElement(static_cast<jobjectArray>(array), i);
+					std::string result = reflector->getField<std::string>(static_cast<jstring>(element));
+					//fprintf(stderr, "write string\n");
+					EIOS_Write(response, ReflectionBinaryType::STRING);
+					//fprintf(stdout, "write string[%zu]\n", i);
+					EIOS_Write(response, result);
+					reflector->getEnv()->DeleteLocalRef(element);
+				}
+			}
+			break;
+
+			case ReflectionArrayType::OBJECT:
+			{
+				for (std::size_t i = 0; i < length; ++i)
+				{
+					auto result = reflector->getEnv()->GetObjectArrayElement(static_cast<jobjectArray>(array), i);
+					//fprintf(stderr, "write obj\n");
+					EIOS_Write(response, ReflectionBinaryType::OBJECT);
+					//fprintf(stdout, "write obj[%zu]\n", i);
+					EIOS_Write(response, result ? reflector->getEnv()->NewGlobalRef(result) : nullptr);
+					reflector->getEnv()->DeleteLocalRef(result);
+				}
+			}
+			break;
+		}
+	};
+
+	std::function<void(jarray, ReflectionArrayType, void*&, int)> process_array;
+	process_array = [this, &write_result, &process_array](jarray array, ReflectionArrayType type, void*& response, int dimensions) -> void
```

Comment:
`auto process_array = ....` No need for the `std::function` assignment because `std::function` can be slow and Lambdas are much faster. So this will be better for sure.

Reply:
mscv doesn't seem to like using auto with recursive calls.